crates/floresta-electrum/src/electrum_protocol.rs

```diff
@@ -576,6 +576,36 @@ impl<Blockchain: BlockchainInterface> ElectrumServer<Blockchain> {
                             .write(serde_json::to_string(&res).unwrap().as_bytes())
                             .await?;
                     }
+                } else if let Ok(requests) = serde_json::from_str::<Vec<Request>>(&msg) {
+                    for req in requests {
+                        let client = self.clients.get(&client);
+                        if client.is_none() {
+                            error!("Client sent a message but is not listed as client");
+                            return Ok(());
+                        }
+                        let client = client.unwrap().to_owned();
+                        let id = req.id.to_owned();
+                        let res = self.handle_client_request(client.clone(), req).await;
+
+                        if let Ok(res) = res {
```

Comment:
In a batch request, you have to respond with an array object, not the idividual ones. E.g:

```json
[{"json-rpc": "2.0", "method": "something", "params": [], "id": 0}, {"json-rpc": "2.0", "method": "something", "params": [], "id": 0}]
```
Should return

```json
[{"id":"0","jsonrpc":"2.0","result": "something"},  {"id":"0","jsonrpc":"2.0","result": "something2"}]
```
and not
```
{"id":"0","jsonrpc":"2.0","result": "something"}
{"id":"0","jsonrpc":"2.0","result": "something2"}
```

Reply:
I modify the code please take a look

Reply:
Looks great! Just one little thing: if we can't parse the json, we need to give an error indicating so. Currently, it only hangs up.

Reply:
@Davidson-Souza  what do I need to do if it is unable to parse the data ??
along with the payload we are receiving the other data also  (not only payload JSON), 
![image](https://github.com/Davidson-Souza/Floresta/assets/97228857/fb6c9935-19ca-4885-b6a4-7b87ed6fc586)

Reply:
I think you can do something like this:

```rust
let requests = match serde_json::from_str::<Vec<Request>>(&msg).or_else(|_| {
       serde_json::from_str::<Request>(&msg).map(|req| vec![req])
}) {
       Ok() => requests,
       Err(e) => {
           // send an error
           return Ok(());
      }
}
// handle all requests in a for or .map
```

Reply:
@Davidson-Souza,  Actually, I don't think we need another else statement.

The `message_receiver` function used here captures the entire HTTP request, including HTTP headers, payload, etc. Let's take a look at this example: when a client sends an API request, the server receives the following data: line by line , [Reciver_function](https://github.com/Davidson-Souza/Floresta/blob/835a1b021cdf3b493412846e6478afec79897de5/crates/floresta-electrum/src/electrum_protocol.rs#L396) is running inside the while loop

Received Data =>  POST / HTTP/1.1 (The server is unable to parse this message, and the client will receive the error message "Invalid JSON." Subsequently, the connection is lost. The server will never attempt to parse the actual payload. which is  valid)
  Received Data =>    Host: 0.0.0.0:50001
 Received Data =>   User-Agent: python-requests/2.31.0
 Received Data =>  Accept-Encoding: gzip, deflate
Received Data =>   Accept: */*
Received Data =>   Connection: keep-alive
Received Data =>  content-type: application/json
Received Data =>  Content-Length: 62

Received Data =>    {
  "jsonrpc": "2.0",
  "method": "method2",
  "params": [],
  "id": 2
} (Actual Payload)

Reply:
This electrum communication isn't over http, but a raw tcp socket with jsons being exchanged. If you hit it with an HTTP client, it'll error.